cpp/example_code/cross-service/topics_and_queues/README.md

```diff
@@ -0,0 +1,170 @@
+
+Content-based deduplication uses a hash of the content as a deduplication ID. If content-based deduplication is not enabled, you must include a deduplication ID with each message.
+
+```
+Enter a name for your SNS topic:
+```
+
+Topic names can have 1-256 characters. They can contain uppercase and lowercase ASCII letters, numbers, underscores, and hyphens. If you chose a FIFO topic, the application automatically adds a “.fifo” suffix, which is required for FIFO topics.
+
+### Create two SQS queues.
+
+Now, configure two SQS queues to subscribe to your topic.
+
+```
+Enter a name for an SQS queue.
+```
+
+Queue names can have 1-80 characters. They can contain uppercase and lowercase ASCII letters, numbers, underscores, and hyphens. If you chose a FIFO topic, the application automatically adds a “.fifo” suffix, which is required for FIFO queues.
+
+
+```
+Would you like to filter messages for "<queue name>.fifo"s subscription to 
+the topic "<topic name>.fifo"?  (y/n)
+```
+
+If you chose FIFO topics, you can add a filter to the queue’s topic subscription. There are many ways to filter a topic. In this example code, you have the option to filter by a predetermined selection of attributes. For more information about filters, see [Message filtering for FIFO topics](https://docs.aws.amazon.com/sns/latest/dg/fifo-message-filtering.html).
+
+
+```
+You can filter messages by one or more of the following "tone" attributes.
+1. cheerful
+2. funny
+3. serious
+4. sincere
+Enter a number (or enter zero to not add anything more)
+```
+
+If you add a filter, you can select one or more “tone” attributes to filter by. When you’re done, enter “0’” to continue.
+
+The application now prompts you to add the second queue. Repeat the previous steps for the second queue.
+
+The following diagram shows the topic and queue options.
+![Diagram of the options](images/fifo_topics_diagram.png)
+
+After you create the topic and subscribe both queues, the application lets you publish messages to the topic.
+
+
+```
+Enter a message text to publish.
+```
+
+All configurations include a message text.
+
+
+```
+Enter a message group ID for this message.
+```
+
+If this is a FIFO topic, then you must include a group ID. The group ID can contain up to 128 alphanumeric characters `(a-z, A-Z, 0-9)` and punctuation `(!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~).`
+For more information about group IDs, see [Message grouping for FIFO topics](https://docs.aws.amazon.com/sns/latest/dg/fifo-message-grouping.html).
+
+
+```
+Enter a deduplication ID for this message.
+```
+
+If this is a FIFO topic and content-based deduplication is not enabled, then you must enter a deduplication ID. The message deduplication ID can contain up to 128 alphanumeric characters `(a-z, A-Z, 0-9)` and punctuation `(!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~).`
+
```

Comment:
The text formatting on the special characters looks a little odd here.

Reply:
I copied this from docs. I am not sure what is odd about it.